src/packet.rs

```diff
@@ -140,6 +161,13 @@ impl Default for Blob {
     }
 }
 
+impl Reset for Blob {
+    fn reset(&mut self) {
+        self.meta = Meta::default();
+        self.data[..BLOB_HEADER_SIZE].copy_from_slice(&[0u8; BLOB_HEADER_SIZE]);
```

Comment:
TODO: probably non-optimal memset

Reply:
annoyingly doesn't seem like rust has a safe one at the moment

Reply:
what's a better memset() look like?